File: src/main/resources/assets/spark_core/js/blueprint_engine/data/BlueprintNode.js
import { Pin } from './BlueprintPin.js';

/**
 * BlueprintNode 数据结构
 * @property id 唯一标识
 * @property type 节点类型
 * @property position 位置坐标 { x, y }
 * @property inputs 输入引脚列表
 * @property outputs 输出引脚列表
 * @property properties 节点属性映射
 */
export class BlueprintNode {
    constructor(id, type, position = { x: 0, y: 0 }, inputs = [], outputs = [], properties = {}) {
        this.id = id;
        this.type = type;
        this.position = position;
        this.inputs = inputs;
        this.outputs = outputs;
        this.properties = properties;
    }
}

File: src/main/kotlin/cn/solarmoon/spark_core/blueprint/data/BlueprintNode.kt
package cn.solarmoon.spark_core.blueprint.data

import cn.solarmoon.spark_core.blueprint.data.Point2D

/**
 * 占位数据类：蓝图节点定义
 * @property id 节点唯一标识
 * @property type 节点类型标识
 */
data class BlueprintNode(
    val id: String = "",
    val type: String = "",
    val position: Point2D = Point2D(),
    val inputs: MutableList<Pin> = mutableListOf(),
    val outputs: MutableList<Pin> = mutableListOf(),
    val properties: Map<String, Any> = emptyMap()
)
